src/processing/ImageUploader.js

```diff
@@ -0,0 +1,28 @@
+var util = require("../utils");
+var LogService = require("../LogService");
+
+class ImageUploader {
+
+    /**
+     * Uploads images from Slack attachments
+     * @param {WebhookBridge} bridge the WebhookBridge bridge
+     */
+    constructor(bridge) {
+        this._bridge = bridge;
+    }
+
+    uploadImages(attachment) {
+        var promises = [];
+        var attrs = ['author_icon', 'footer_icon', 'image_url']
+        attrs.forEach(key => {
+            var imageUrl = attachment[key]
+            if (imageUrl)
+                var uploadPromise = Promise.resolve(imageUrl);
+                uploadPromise = util.uploadContentFromUrl(this._bridge, imageUrl, this._bridge.getBotIntent());
```

Comment:
These lines are indented incorrectly

Reply:
Are they? Isn't it just `{}` missing?

Reply:
I think the ES6 syntax is confusing me because I think I'm doing coffeescript.
These 3 lines all are supposed to be in the `if`
```
var uploadPromise = Promise.resolve(imageUrl);
uploadPromise = util.uploadContentFromUrl(this._bridge, imageUrl, this._bridge.getBotIntent());
promises.push(uploadPromise.then(mxcUrl => { attachment[key] = mxcUrl }));
```

Reply:
It looks like the first assignment isn't used, so it can be shortened by a line (which is why I initially thought it was wrong indented). 

```
var uploadPromise = util.uploadContent......
// etc
```

Reply:
Ah yes you're right